packages/ui/src/accounts/components/AccountItem/components/lockItems/CouncilorLockItem.tsx

```diff
@@ -29,7 +31,18 @@ export const CouncilorLockItem = ({ lock, address, isRecoverable }: LockDetailsP
       network: eventData.electedAtNetwork,
     })
 
-  const recoveryTime = faker.date.soon(1).toISOString()
+  const recoveryTime = useMemo(() => {
+    if (!eventData || !api) {
+      return null
+    }
+    const startTime = new Date(eventData.electedAtTime).getTime()
+    const idleDuration = api.consts.council.idlePeriodDuration
+    const electionDuration = api.consts.referendum.voteStageDuration?.add(api.consts.referendum.revealStageDuration)
+    const duration = idleDuration.add(electionDuration).toNumber() * MILLISECONDS_PER_BLOCK
```

Comment:
In my understanding the stake is retrievable as soon as the idle period ends. If so this should be enough:
```suggestion
    const startTime = Date.parse(eventData.electedAtTime)
    const duration = api.consts.council.idlePeriodDuration.toNumber()
```